### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by loonytoons**

### CODE REVIEW

File: tests/AnnotationScannerTest.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    protected $app;
+
+    public function setUp(): void
+    {
+        $this->app = m::mock('Illuminate\Contracts\Foundation\Application');
+    }
+
+    public function tearDown(): void
+    {
+        m::close();
```
Comment (jpcaparas) [inline tests/AnnotationScannerTest.php]: Do we still need `Mockery::close()` here? I thought `tearDown` already did that automatically for us?

File: tests/AnnotationScannerTest.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+use Mockery as m;
+
+class AnnotationScannerTest extends TestCase
+{
+    /**
+     * @var m\LegacyMockInterface|m\MockInterface|Illuminate\Contracts\Foundation\Application
+     */
+    protected $app;
+
+    public function setUp(): void
```
Comment (jpcaparas) [inline tests/AnnotationScannerTest.php]: I don't recommend changing the access level of `setUp` and `tearDown`. Let's keep them `protected`.

File: tests/AnnotationScannerTest.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+/**
+ * Class TestAnnotationScanner
+ *
+ * Override class to expose the protected getClassesToScan method
+ */
+class TestAnnotationScanner extends AnnotationScanner
+{
+    public function getClassesToScan()
+    {
+        return parent::getClassesToScan();
```
Comment (jpcaparas) [inline tests/AnnotationScannerTest.php:87]: Uhm, I'm not sure what this is for. Do we really need it?

File: tests/AnnotationScannerTest.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        require_once __DIR__.'/Events/fixtures/annotations/BasicEventHandler.php';
+
+        $scanner = new TestAnnotationScanner([]);
+        $scanner->setClassesToScan([
+            'App\Handlers\Events\BasicEventHandler',
+            'App\Handlers\Events\NonExistentHandler',
+        ]);
+
+        $classesToScan = $scanner->getClassesToScan();
+        $this->assertCount(1, $classesToScan);
```
Comment (jpcaparas) [inline tests/AnnotationScannerTest.php:73]: Let's put in messages on the third arguments of `assertCount` and `assertEquals` to give the dev a friendly message of what may have went wrong.

Comment (jpcaparas) [reply tests/AnnotationScannerTest.php:73]: Probably goes with other `assert-` calls as well.

Comment (jpcaparas) [reply tests/AnnotationScannerTest.php:73]: e.g.

    $this->assertCount(1, $classesToScan, 'Classes to scan exceeded expectations.');

File: tests/AnnotationFinderTest.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+     * @covers \Collective\Annotations\AnnotationFinder::eventsAreScanned
+     * @covers \Collective\Annotations\AnnotationFinder::getScannedEventsPath
+     * @covers \Collective\Annotations\AnnotationFinder::modelsAreScanned
+     * @covers \Collective\Annotations\AnnotationFinder::getScannedModelsPath
+     */
+    public function testRoutesAreScannedReturnsTrue($filepath, $found, $method, $expected)
+    {
+        $this->app->shouldReceive('make')
+            ->with('files')->once()
+            ->andReturn($files = m::mock('Illuminate\Filesystem\Filesystem'));
```
Comment (jpcaparas) [inline tests/AnnotationFinderTest.php]: Do we have other behaviour to expect on `m::mock('Illuminate\Filesystem\Filesystem')`?

File: tests/AnnotationFinderTest.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
+<?php
+
+use Collective\Annotations\AnnotationFinder;
+use PHPUnit\Framework\TestCase;
+use Mockery as m;
+
+class AnnotationFinderTest extends TestCase
+{
+    /**
+     * @var m\LegacyMockInterface|m\MockInterface|Illuminate\Contracts\Foundation\Application
```
Comment (jpcaparas) [inline tests/AnnotationFinderTest.php]: Do we still really need `LegacyMockInterface`? Will it return that at all?

File: tests/AnnotationScannerTest.php [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        Container::setInstance($this->app);
+
+        App::shouldReceive('make')
+            ->with('path')->once()
+            ->andReturn('src');
+
+        $scanner = new Scanner(['App\Handlers\Events\BasicEventHandler']);
+
+        $return = $scanner->addAnnotationNamespace(
+            'App\Events\Annotations\Annotations'
```
Comment (jpcaparas) [inline tests/AnnotationScannerTest.php:49]: Would it make sense to use `Annotations::class` here over `App\Events\Annotations\Annotations`?

File: tests/AnnotationScannerTest.php (hunk 3 above)
Comment (loonytoons) [reply tests/AnnotationScannerTest.php:87]: this was to make it publicly accessible

File: tests/AnnotationScannerTest.php (hunk 2 above)
Comment (loonytoons) [reply tests/AnnotationScannerTest.php]: doh, this was unintentional

File: tests/AnnotationScannerTest.php (hunk 1 above)
Comment (loonytoons) [reply tests/AnnotationScannerTest.php]: Only if you extend the MockeryTestCase or include the mockery adapter trait. I'll add the latter

File: tests/AnnotationScannerTest.php (hunk 4 above)
Comment (loonytoons) [reply tests/AnnotationScannerTest.php:73]: I only usually do this when it's not super obvious based on the line of the test file what's gone wrong

File: tests/AnnotationFinderTest.php (hunk 5 above)
Comment (loonytoons) [reply tests/AnnotationFinderTest.php]: What are you thinking of? The next code block sets that up.
Or is it confusing that it's assigned here? And then the assertions added? I'll move it around

File: tests/AnnotationFinderTest.php (hunk 6 above)
Comment (loonytoons) [reply tests/AnnotationFinderTest.php]: I doubt it, that was just auto added based on the underlying code. I can take it out though

File: tests/AnnotationScannerTest.php (hunk 7 above)
Comment (loonytoons) [reply tests/AnnotationScannerTest.php:49]: No as this isn't actually a class. It's a namespace that also doesn't technically exist, but due to how the test is set up, this namespace is mapped to the `Events/Annotations/Annotations` dir which we use to actually test the scanner. 
I'll add some more comments to the test, as it is a bit confusing

File: tests/AnnotationScannerTest.php (hunk 4 above)
Comment (loonytoons) [reply tests/AnnotationScannerTest.php:73]: @jpcaparas I haven't added this, but let me know if it's something that we really do want. I'll add it later en-masse if so

**Comment by loonytoons** [comment]
Thanks for the review @jpcaparas 🙌

**Pull request merged**